lib/index.ts

```diff
@@ -184,38 +203,25 @@ export default class Backburner {
 
   public run(method: Function);
   public run(target: Function | any | null, method?: Function | string, ...args);
-  public run(target: any | null | undefined, method?: any, ...args: any[]) {
-    let length = arguments.length;
-    let _method: Function | string;
-    let _target: any | null | undefined;
-
-    if (length === 1) {
-      _method = <Function> target;
-      _target = null;
-    } else {
-      _method = method;
-      _target = target;
-
-      if (isString(_method)) {
-        _method = <Function> _target[_method];
-      }
-    }
+  public run(target: any | null | undefined, method?: any, ...args: any[]);
+  public run(..._args) {
```

Comment:
I think we probably only want to use `..._args` for the `schedule` case (the ones that require `queueName` as the first arg, but have the same remaining signature).

Doing `...args` here forces the transpiled code to iterate over all of the arguments [like this](http://www.typescriptlang.org/play/index.html#src=class%20Foo%20%7B%0A%0A%20%20run(..._args)%20%7B%0A%20%20%20%20someMethod(_args)%3B%0A%20%20%7D%0A%7D):

```js
        var _args = [];
        for (var _i = 0; _i < arguments.length; _i++) {
            _args[_i] = arguments[_i];
        }
        someMethod(_args);
```

But in the `run` and `run.join` case, it isn't needed.  

Changing this to be (and updating `parseArgs` to use `arguments` instead of a single passed in array):

```js
run() {
  let [ target, method, args ] = parseArgs(...arguments);
  // ...snip...
}
```

Then for `schedule` you can do:

```js
schedule(queueName, ...args) {
  let [ target, method, args ] = parseArgs(...args);
}
```

Reply:
with just single run of `npm run bench` both ways (suggested and already done) performing equally, because `parseArgs(...args);` will transpile to `parseArgs.appy(null, args)` and in `parseArgs` I will have to work with `arguments` instead of `array` which is as bad as for loop with arguments I think

Reply:
> because parseArgs(...args); will transpile to parseArgs.appy(null, args)

Yes, but every `run`, `run.join`, `schedule`, etc will result in an extra array allocation (two actually, but one of them is avoidable)...

> I will have to work with arguments instead of array which is as bad as for loop with arguments I think

Accessing arguments by index should not be any slower than the array operations that exist in the PR currently, and is what we are doing now in current released versions.

Reply:
didn't think about array allocation, pushed as you suggested now 👍